srcpkgs/ncspot/template

```diff
@@ -0,0 +1,35 @@
+# Template file for 'ncspot'
+pkgname=ncspot
+version=0.11.1
+revision=1
+build_style=cargo
+configure_args="--no-default-features"
+hostmakedepends="pkg-config python3"
+makedepends="openssl-devel $(vopt_if libxcb libxcb-devel)
+ $(vopt_if dbus dbus-devel) $(vopt_if alsa alsa-lib-devel)
+ $(vopt_if pulseaudio pulseaudio-devel)"
+short_desc="Cross-platform ncurses Spotify client"
+maintainer="Omar Zeghouani <omarzeghouanii@gmail.com>"
+license="BSD-2-Clause"
+homepage="https://github.com/hrkfdn/ncspot"
+distfiles="https://github.com/hrkfdn/ncspot/archive/v${version}.tar.gz"
+checksum=cfd8d0ea65321652cd00c91b47aaa52bd073bc4afa7e971b8f0e84e0f80eb9ff
+
+build_options="libxcb dbus alsa pulseaudio notify cover"
```

Comment:
why do these all need to be build options?

Reply:
Maybe because all are **features**, https://github.com/hrkfdn/ncspot/blob/main/Cargo.toml

Reply:
yes I'm aware of that, but it would be best to choose a sane default set of features and maybe have one or two settable via build options. void isn't gentoo and we don't need to give the option for everything

Reply:
I'm the package maintainer on NetBSD and I've decided to default to 'portaudio' and 'termion'.

On Void, I'd go with 'alsa', haven't managed to get 'rodio' working on musl (haven't tried that hard to be honest), and 'termion', cause I don't like to force 'ncurses'.

That said, this are just my prefences.

Reply:
> why do these all need to be build options?

they don't. I decided to just use the default features that the ncspot maintainer set. 
https://github.com/hrkfdn/ncspot/blob/7e974bb7ceeeca50dc32e77209dd113f86c9fc1c/Cargo.toml#L76

> void isn't gentoo and we don't need to give the option for everything

Removed the optional build feaures

Reply:
> void isn't gentoo and we don't need to give the option for everything

https://github.com/void-linux/void-packages/blob/2b4937e7eb0a194dfb5a4bdbce438b657f94f705/srcpkgs/mpv/template#L6-L36

Reply:
> Removed the optional build features

i'm not saying you need to remove them outright, just that you don't need to make them build options. it's fine to select some/all of them to build by default

Reply:
I reviewed the default build features and they seem sane. I wasn't using the optional features myself tbh.

Reply:
> I decided to just use the default features that the ncspot maintainer set.

it would be reasonable to add in alsa_backend and cover i think

Reply:
Added both alsa_backend and cover